### PR TITLE
Remove AssertPermissionTrait

### PR DESCRIPTION
**Pull request by tankerkiller125**
Removes AssertPermissionTrait

### CODE REVIEW

File: src/Command/CreateFlagHandler.php [hunk 1]
```diff
@@ -67,7 +64,7 @@ public function handle(CreateFlag $command)
             throw new InvalidParameterException;
         }
 
-        $this->assertCan($actor, 'flag', $post);
+        $actor->can('flag', $post);
```
Comment (askvortsov1) [inline src/Command/CreateFlagHandler.php]: This should be $actor->assertCan

File: src/Command/DeleteFlagsHandler.php [hunk 2]
```diff
@@ -49,7 +46,7 @@ public function handle(DeleteFlags $command)
 
         $post = $this->posts->findOrFail($command->postId, $actor);
 
-        $this->assertCan($actor, 'viewFlags', $post->discussion);
+        $actor->can('viewFlags', $post->discussion);
```
Comment (askvortsov1) [inline src/Command/DeleteFlagsHandler.php]: This should be $actor->assertCan

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/Api/Controller/ListFlagsController.php [hunk 3]
```diff
@@ -42,7 +39,7 @@ protected function data(ServerRequestInterface $request, Document $document)
     {
         $actor = $request->getAttribute('actor');
 
-        $this->assertRegistered($actor);
+        ! $actor->isGuest();
```
Comment (dsevillamartin) [inline src/Api/Controller/ListFlagsController.php]: This won't do anything?

Comment (tankerkiller125) [reply src/Api/Controller/ListFlagsController.php]: Fixed in newest commit

**Review by franzliedke — APPROVED** [review]

**Pull request merged**